pkgs/development/python-modules/pytorch-lightning/default.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, isPy3k
+, scikitlearn
+, tqdm
+, numpy
+, pytorch
+, torchvision
+, pandas
+, test-tube
+, future
+}:
+
+buildPythonPackage rec {
+  pname = "pytorch-lightning";
+  version = "0.5.3.2";
+
+  disabled = !isPy3k; # requires python version >=3.6
+
+  src = fetchFromGitHub {
+    owner = "williamFalcon";
+    repo = pname;
+    rev = version;
+    sha256 = "0vws85dkw2df243fgsh7c8vz9l33a57gvf6lchzb0h2v16q4h2hm";
+  };
+
+
+  # comet_ml not currently available in nixpkgs
+  # not required for pytorch-lightning
+  patchPhase = ''
+    substituteInPlace requirements.txt \
+      --replace "tqdm==4.35.0" "tqdm" \
+      --replace "numpy==1.16.4" "numpy" \
+      --replace "scikit-learn==0.20.2" "scikit-learn" \
+      --replace "torchvision>=0.3.0" "torchvision"
+    substituteInPlace pytorch_lightning/logging/__init__.py \
+      --replace "from .comet_logger import CometLogger" ""
+      rm pytorch_lightning/logging/comet_logger.py
+  '';
+    
+  propagatedBuildInputs = [
+    scikitlearn
+    tqdm
+    numpy
+    pytorch
+    torchvision
+    pandas
+    test-tube
+    future
+  ];
+  
+  # lots of network access needed: https://gist.github.com/09b804918e5ca4b7dc55bb28ca4597f5
+  doCheck = false;
+
+  # checkPhase = ''
+  #   coverage run --source pytorch_lightning -m py.test pytorch_lightning tests pl_examples -v --doctest-modules
```

you don't have to run all the tests
```suggestion
  py.test pytorch_lightning tests -k 'not this_network_test'
```